crates/swift-bridge-ir/src/bridged_type.rs

```diff
@@ -86,23 +86,23 @@ pub(crate) trait BridgeableType: Debug {
     /// # Examples
     /// String -> None
     /// Result<String, OpaqueRust> -> None
-    /// Result<(), TransparentEnum> -> pub enum ResultVoidAndTransparentEnum { //... }
-    fn generate_custom_rust_ffi_type(
+    /// Result<(), TransparentEnum> -> Some(vec![pub enum ResultVoidAndTransparentEnum { //... }])
+    fn generate_custom_rust_ffi_types(
         &self,
         swift_bridge_path: &Path,
         types: &TypeDeclarations,
-    ) -> Option<TokenStream>;
+    ) -> Option<Vec<TokenStream>>;
```

Comment:
>Right now It think we should generate the FFI types for (u32, u32) twice

Custom Definitions is managed by `HashMap`, so it cannot be generated twice. 

https://github.com/chinedufn/swift-bridge/blob/9e09fcefbd2581fc6e44cb4207c99d25eb175b4d/crates/swift-bridge-ir/src/parsed_extern_fn.rs#L158-L179

Reply:
Here's a better example:

```rust
#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        fn a() -> Result<(u32, u32), ()>
        fn b() -> (u32, u32)
    }
}
```

I think this would generate the `(u32, u32)` support types twice. (as you described in your other comment)

---

So, yeah, this works for now but will have to change in the future.

I still think `-> TokenStream` makes more sense than `-> Vec<TokenStream>` , especially since what we're really doing is generating support code, not just types.

Let me know if you disagree.

This is all pretty minor stuff since this is all used internally, so not a huge deal, just calling this out since changing the signature seems unnecessary and less clear to me.

Reply:
Do you have thoughts?

Reply:
I apologize for the late response. I'll write my opinion later.

Reply:
Maybe we need to add a unit test for this error to this PR.

Reply:
I'm not understanding what you're saying?

I'm just talking about returning `->TokenStream` instead of `-> Vec<TokenStream>`.

So, I'm essentially saying to concatenated the `Vec<TokenStream>` that you're creating into a single `TokenStream`.

Is what you're saying related to that?

Reply:
>I'm just talking about returning ->TokenStream instead of -> Vec<TokenStream>.

The custom types are managed by a `HashMap<String, TokenStream>`. If you return a `TokenStream`, the contents of the `HashMap<String, TokenStream>` will be something like:

key1: `"# [repr (C)] pub enum ResultTupleI32U32AndSomeErrEnum { Ok (__swift_bridge__tuple_I32U32) , Err (__swift_bridge__SomeErrEnum) , }, # [repr (C)] # [doc (hidden)] pub struct __swift_bridge__tuple_I32U32 (i32 , u32) ;"`
value1: `# [repr (C)] pub enum ResultTupleI32U32AndSomeErrEnum { Ok (__swift_bridge__tuple_I32U32) , Err (__swift_bridge__SomeErrEnum) , }, # [repr (C)] # [doc (hidden)] pub struct __swift_bridge__tuple_I32U32 (i32 , u32) ;`

key1: `"# [repr (C)] # [doc (hidden)] pub struct __swift_bridge__tuple_I32U32 (i32 , u32) ;"`
value1: `# [repr (C)] # [doc (hidden)] pub struct __swift_bridge__tuple_I32U32 (i32 , u32) ;`

Due to the double definition of `__swift_bridge__tuple_I32U32`, the error occurs, like so:
```
the name `__swift_bridge__tuple_I32ResultTestOpaqueRustTypeString` is defined multiple times
`__swift_bridge__tuple_I32ResultTestOpaqueRustTypeString` must be defined only once in the type namespace of this module
```

 This problem can be solved by individually managing the custom types in the `HashMap<String, TokenStream>`. So, we need to return a `Vec<TokenStream>`. If we return a `Vec<TokenStream>`, the contents of the `HashMap<String, TokenStream>` will be something like:

key1: `"# [repr (C)] pub enum ResultTupleI32U32AndSomeErrEnum { Ok (__swift_bridge__tuple_I32U32) , Err (__swift_bridge__SomeErrEnum) , }"`
value1: `# [repr (C)] pub enum ResultTupleI32U32AndSomeErrEnum { Ok (__swift_bridge__tuple_I32U32) , Err (__swift_bridge__SomeErrEnum) , }`

key2: `"# [repr (C)] # [doc (hidden)] pub struct __swift_bridge__tuple_I32U32 (i32 , u32) ;"`
value2: `# [repr (C)] # [doc (hidden)] pub struct __swift_bridge__tuple_I32U32 (i32 , u32) ;`

Reply:
Here's an example of using `HashMap<String, TokenStream>`

https://github.com/chinedufn/swift-bridge/blob/9e09fcefbd2581fc6e44cb4207c99d25eb175b4d/crates/swift-bridge-ir/src/parsed_extern_fn.rs#L158-L179

Reply:
Gotcha, thanks for explaining.

I have a feeling that we could make this a bit more clear, but this works for now since it's an internal detail.

Reply:
>I have a feeling that we could make this a bit more clear

I think so too, but I have no idea now. It might be good to create an issue to discuss this further.